catalog/spatial/registry/registry-report-action-provider/src/main/java/org/codice/ddf/registry/report/action/provider/RegistryReportActionProvider.java

```diff
@@ -81,19 +81,14 @@ public RegistryReportActionProvider(String id) {
 
     private List<Action> processSubject(Metacard metacard) {
 
-        if (metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID) == null
-                || StringUtils.isBlank(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID)
-                .getValue()
-                .toString())) {
+        if (!RegistryUtility.hasRegistryId(metacard)) {
```

Comment:
Same kind of question as before. Can we getRegistryId and check for null. Then use that in the encode. Or create a separate hasRegistryId that doesn't care about registry tag. You may already be doing something like that since you're renaming hasRegistryId anyway.